src/widgets/CalendarWidget.js

```diff
@@ -193,7 +193,10 @@ export default class CalendarWidget extends InputWidget {
               // Make sure we commit the value after a blur event occurs.
               this.addEventListener(this.calendar._input, 'blur', (event) => {
                 if (!event.relatedTarget?.className.split(/\s+/).includes('flatpickr-day')) {
-                  this.calendar.setDate(new Date(this.calendar.input.value), true, this.settings.altFormat);
+                  const inputValue = this.calendar.input.value;
+                  const dateValue = inputValue ? moment(this.calendar.input.value, convertFormatToMoment(this.valueFormat)).toDate() : inputValue;
```

Comment:
I just replaced new Date( ) with moment() here. The problem is: new Date() returns 'Invalid Date'/wrong date for custom formats (like 'dd-MM-yyyy' and other) as well as for '', null and undefined. That is why I replaced it with moment(). If just moment(value).toDate() without format argument is used, it behaves in the same way as new Date( ) and returns 'Invalid date'/wrong date for some formats. To identify the date correctly, moment() needs second argument, namely the format which the first argument (date value) complies with.
Also here I added a check, if an inputValue is empty, there is no need to convert it into date using moment() to avoid getting 'Invalid Date'